### ISSUE TITLE
[question] Indirect dependencies missing when using PkgConfigDeps

### ISSUE THREAD

**Issue by joastoe** [conan-io/conan#9939]
There's an issue regarding **pkg config** I'm fighting with:

I have libA->libB->libC

- libA requires libB
- libB requires libC

all regular `requires` not `build_requires`

When building libA with the pkg_config generator I get the following generated .pc file for libB
```
prefix=/.conan/data/libB/1.0.0_test/company/testing/package/13ca7c3421ced37746801df275efb2bb7f8fca28
libdir=${prefix}/lib
includedir=${prefix}/include

Name: libB
Description: Library B
Version: 1.0.0_test
Libs: -L"${libdir}" -lds  -lds-tests  -Wl,-rpath,"${libdir}"
Cflags: -I"${includedir}"
Requires: libC
```

After switching to PkgConfigDeps, the generated .pc file changes:
```
prefix=/.conan/data/libB/1.0.0_test/company/testing/package/13ca7c3421ced37746801df275efb2bb7f8fca28
libdir=${prefix}/lib
includedir=${prefix}/include

Name: libB
Description: Library B
Version: 1.0.0_test
Libs: -L"${libdir}" -lds  -lds-tests  -Wl,-rpath,"${libdir}"
Cflags: -I"${includedir}"
```
The **Requires** attribute is missing for the indirect dependency.
Can someone help me out with this?

Thanks very much

**Comment by memsharded** [comment]
Fixed by @franramirez688 in https://github.com/conan-io/conan/pull/9945, it will be released in next 1.43.
Thanks very much for reporting!

**Comment by joastoe** [comment]
Very nice, thanks for fixing. Looking forward to 1.43.
